src/ApiManagement/ApiManagement.ServiceManagement/Commands/RemoveAzureApiManagementApiFromGateway.cs

```diff
@@ -0,0 +1,63 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementApiFromGateway", SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzureApiManagementApiFromGateway : AzureApiManagementCmdletBase
```

Comment:
Please support pipe for this cmdlet.

Reply:
Hi @wyunchi-ms , could you please provide more details what does "support pipe" mean?
Thanks you.

Reply:
@vfedonkin , sorry for missing the example. You can see the examples [here](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/remove-cmdlet-example.md).

Reply:
@wyunchi-ms ,  sorry, need more details, what's exactly missing.
You mean, we need to add support for these additional parameters? :

[Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = DeleteByInputObjectParameterSet)]
        [ValidateNotNull]
        public PSTopLevelResource InputObject { get; set; }

        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = DeleteByResourceIdParameterSet)]
        [ValidateNotNullOrEmpty]
        public string ResourceId { get; set; }

Reply:
@wyunchi-ms , we don't have any GET for this resource. It's actually not a resource, it's an assignment between two independent resources: API and Gateway may be linked, but not like "parent-child".
So we have like two "parent resources" for this assignment:  api+gateway. Pipeline not supported for this scenario.